src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -12,68 +12,297 @@
+        newQnForm.setCourseId(feedbackSession.courseId);
+        newQnForm.setFeedbackSessionName(feedbackSession.feedbackSessionName);
+        newQnForm.setQuestionNumberSuffix("");
+        
+        newQnForm.setQuestionTypeOptions(getQuestionTypeChoiceOptions());
+      
+        FeedbackQuestionFeedbackPathSettings feedbackPathSettings = new FeedbackQuestionFeedbackPathSettings();
+        FeedbackQuestionVisibilitySettings visibilitySettings = new FeedbackQuestionVisibilitySettings();
+        
+        newQnForm.setFeedbackPathSettings(feedbackPathSettings);
+        newQnForm.setVisibilitySettings(visibilitySettings);
+        
+        feedbackPathSettings.setGiverParticipantOptions(getParticipantOptions(null, true));
+        feedbackPathSettings.setRecipientParticipantOptions(getParticipantOptions(null, false));
+        feedbackPathSettings.setNumOfEntitiesToGiveFeedbackToValue(1);
+        
+        newQnForm.setNumOfQuestionsOnPage(questions.size());
+        newQnForm.setQuestionNumberOptions(getQuestionNumberOptions(questions.size() + 1));
+      
+        newQnForm.setQuestionSpecificEditFormHtml(getNewQuestionSpecificEditFormHtml());
+        newQnForm.setEditable(true);
+        
+        setDefaultVisibilityOptions(visibilitySettings, feedbackPathSettings);
+    }
+
+
+    private void setDefaultVisibilityOptions(FeedbackQuestionVisibilitySettings visibilityOptions,
+                                             FeedbackQuestionFeedbackPathSettings feedbackPathSettings) {
+        Map<String, Boolean> isGiverNameVisible = new HashMap<String, Boolean>();
+        Map<String, Boolean> isRecipientNameVisible = new HashMap<String, Boolean>();
+        Map<String, Boolean> isResponsesVisible = new HashMap<String, Boolean>();
+        
+        List<FeedbackParticipantType> participantVisiblity = new ArrayList<FeedbackParticipantType>();
+        participantVisiblity.add(FeedbackParticipantType.INSTRUCTORS);
+        participantVisiblity.add(FeedbackParticipantType.RECEIVER);
+        
+        for (FeedbackParticipantType participant : participantVisiblity) {
+            isGiverNameVisible.put(participant.name(), true);
+            isRecipientNameVisible.put(participant.name(), true);
+            isResponsesVisible.put(participant.name(), true);
+        }
+        
+        visibilityOptions.setGiverNameVisibleFor(isGiverNameVisible);
+        visibilityOptions.setRecipientNameVisibleFor(isRecipientNameVisible);
+        visibilityOptions.setResponseVisibleFor(isResponsesVisible);
+    }
+    
+    
+    private void buildBasicFsForm(FeedbackSessionAttributes newFeedbackSession, FeedbackSessionsAdditionalSettingsFormSegment additionalSettings) {
+        fsForm = FeedbackSessionsForm.getFsFormForExistingFs(this, newFeedbackSession, additionalSettings);
+    }
+
+    private FeedbackSessionsAdditionalSettingsFormSegment buildFsFormAdditionalSettings(FeedbackSessionAttributes newFeedbackSession) {
```

Comment:
How about moving these methods up, to around the method that calls them?
It'll make it easier to reference since it'll be possible to have all of them on-screen at the same time (due to their lengths).